core/dpdk.cc

```diff
@@ -108,16 +118,21 @@ void init_eal(int dpdk_mb_per_socket, std::string nonworker_corelist) {
       // Do not bother with /var/run/.rte_config and .rte_hugepage_info,
       // since we don't want to interfere with other DPDK applications.
       "--no-shconf",
+      // TODO(sangjin) switch to dynamic memory mode
+      "--legacy-mem",
   };
 
   if (dpdk_mb_per_socket <= 0) {
+    rte_args.Append({"--iova", "va"});
     rte_args.Append({"--no-huge"});
 
     // even if we opt out of using hugepages, many DPDK libraries still rely on
     // rte_malloc (e.g., rte_lpm), so we need to reserve some (normal page)
     // memory in advance. We allocate 512MB (this is shared among nodes).
     rte_args.Append({"-m", "512"});
   } else {
+    rte_args.Append({"--iova", "pa"});
```

Comment:
what are the capabilities needed?
does it need `iommu=pt`? today we use `intel_iommu=on` only.

Reply:
@chris3torek i wanted to know why we should force PA and indirectly SYS_ADMIN when using hugepages? Today we use hugepages without needing root.